### PR TITLE
MOSIP 43544: Automated testcases for GET/admin-partners With Status Query Parameter.

### PR DESCRIPTION
**Pull request by SradhaMohanty5899**
Added automated test cases for the GET /admin-partners API to validate partner filtering using the status query parameter and verify correct response behavior.

## Summary by CodeRabbit

* **Tests**
  * Expanded partner-listing coverage with status filters (active, inactive, deactivated), pagination, sorting, certificates, policy-group interactions, partner type/id scenarios, and numerous edge cases (invalid/empty/special-char statuses).
  * Removed a single error-only check and added many new test cases using input/output templates.
* **Templates**
  * Added input and output templates to standardize request payloads and response assertions.
* **Test Configuration**
  * Updated a test mapping to use a query-parameter test runner variant.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: api-test/src/main/resources/pms/GetListOfPartners/GetListOfPartners.yml [hunk 1]
```diff
@@ -527,4 +526,239 @@ GetListOfPartners:
...
+      restMethod: get
+      checkErrorsOnlyInResponse: true
+      inputTemplate: pms/GetListOfPartners/getListOfPartners
+      outputTemplate: pms/GetListOfPartners/GetListOfPartnersResult
+      input: '{
+}'
+      output: '{
+      "pageNo": "0"
+}'
+   Pms_GetListOfPartners_With_Empty_Status_Neg:
```
Comment (mohanachandran-s) [inline api-test/src/main/resources/pms/GetListOfPartners/GetListOfPartners.yml:724]: For negative scenarios. totalResults has to be validated for 0

**Review by mohanachandran-s — CHANGES_REQUESTED** [review]

**Review by mohanachandran-s — APPROVED** [review]

**Review by mayuradesh — APPROVED** [review]
